fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/MessageValidator.kt

```diff
@@ -219,6 +234,54 @@ object MessageValidator {
       return this
     }
 
+    /**
+     * Helper method to check a list of questions for validity.
+     *
+     * @param questions the list of questions to check.
+     * @throws IllegalArgumentException if any question does not meet the criteria.
+     */
+    fun validQuestions(questions: List<ElectionQuestion.Question>?): MessageValidatorBuilder {
+      require(!questions.isNullOrEmpty()) { "Questions list cannot be null or empty" }
+      noListDuplicates(questions)
+      for (question in questions) {
+        validQuestion(question.title, question.votingMethod, question.ballotOptions)
+      }
+      return this
+    }
+
+    /**
+     * Helper method to check a single question for validity.
+     *
+     * @param title the title of the question
+     * @param votingMethod the voting method of the question
+     * @param ballotOptions the ballot options of the question
+     * @throws IllegalArgumentException if the question does not meet the criteria.
+     */
+    fun validQuestion(
+        title: String,
+        votingMethod: String,
+        ballotOptions: List<String>,
+    ): MessageValidatorBuilder {
+      stringNotEmpty(title, "question title")
+      val validVotingMethods = ElectionSetupFragment.VotingMethods.values().map { it.desc }
+      require(votingMethod in validVotingMethods) {
+        "Unsupported voting method in question: ${title}. Must be one of $validVotingMethods."
+      }
+      validBallotOptions(ballotOptions)
+      return this
+    }
+
+    private fun validBallotOptions(ballotOptions: List<String>?): MessageValidatorBuilder {
+      requireNotNull(ballotOptions) { "Ballot options cannot be null" }
+      listNotEmpty(ballotOptions)
+      require(ballotOptions.size >= 2) { "There must be at least 2 ballot options" }
+      noListDuplicates(ballotOptions)
+      ballotOptions.map {
+        stringNotEmpty(it, "ballot option in place " + ballotOptions.indexOf(it))
+      }
```

Comment:
I don't actually get the usage of map here, as we don't actually want to map the result of the verification back in the list. I'd rather use a forEach() instead